src/components/Time/ExpiredTimestepManager.vue

```diff
@@ -50,69 +56,141 @@ export default {
-          );
-          this.expiredSnackBarMessage = this.$t("ExpiredExtentRefreshed");
+        this.expiredTimestepList = [];
+        if (noChangeFlag) {
+          this.expiredSnackBarMessage = this.$t("MissingTimesteps");
           this.notifyExtentRebuilt = true;
           if (this.isAnimating) {
             this.$root.$emit("redoAnimation");
+          } else {
+            this.$root.$emit("fixLayerTimes");
           }
         } else {
-          const currentHighBoundDate =
-            this.getMapTimeSettings.Extent[this.datetimeRangeSlider[1]];
-          this.changeMapTime(this.getMapTimeSettings.Step, null, 0);
-          if (this.getMapTimeSettings.Extent[0] >= currentHighBoundDate) {
-            // Cancel animation
-            this.expiredSnackBarMessage = this.$t("extentFullyOOB");
+          if (this.getMapTimeSettings.SnappedLayer !== null) {
+            this.changeMapTime(
+              this.getMapTimeSettings.Step,
+              this.$mapLayers.arr.find(
+                (l) =>
+                  l.get("layerName") === this.getMapTimeSettings.SnappedLayer
+              )
+            );
+            this.expiredSnackBarMessage = this.$t("MissingTimesteps");
             this.notifyExtentRebuilt = true;
-            this.$store.commit("Layers/setDatetimeRangeSlider", [
-              0,
-              this.getMapTimeSettings.Extent.length - 1,
-            ]);
             if (this.isAnimating) {
-              this.$root.$emit("restoreState");
+              this.$root.$emit("redoAnimation");
             }
           } else {
-            const highBoundIndex =
-              this.datetimeRangeSlider[1] - this.datetimeRangeSlider[0];
-            this.expiredSnackBarMessage = this.$t("ExpiredExtentRefreshed");
-            this.notifyExtentRebuilt = true;
-            this.$store.commit("Layers/setDatetimeRangeSlider", [
-              0,
-              highBoundIndex,
-            ]);
-            if (this.isAnimating) {
-              this.$root.$emit("redoAnimation");
+            const currentHighBoundDate =
+              this.getMapTimeSettings.Extent[this.getDatetimeRangeSlider[1]];
+            this.changeMapTime(this.getMapTimeSettings.Step);
+            if (this.getMapTimeSettings.Extent[0] >= currentHighBoundDate) {
+              // Cancel animation
+              this.expiredSnackBarMessage = this.$t("ExtentFullyOOB");
+              this.notifyExtentRebuilt = true;
+              if (this.isAnimating) {
+                this.$root.$emit("restoreState");
+              }
+            } else {
+              this.expiredSnackBarMessage = this.$t("MissingTimesteps");
+              this.notifyExtentRebuilt = true;
+              if (this.isAnimating) {
+                this.$root.$emit("redoAnimation");
+              }
             }
           }
         }
       }
+      this.$root.$emit("loadingStop");
+    },
+    async errorDispatcher(layer, e) {
+      var xhttp = new XMLHttpRequest();
```

Comment:
Why not use Axios?

Reply:
Because I felt like making it more complicated for no reason, changed to axios